internal/operator-controller/rukpak/convert/render/generate.go

```diff
@@ -0,0 +1,191 @@
+	clusterPermissions := rv1.CSV.Spec.InstallStrategy.StrategySpec.ClusterPermissions
+
+	// If we're in AllNamespaces mode, promote the permissions to clusterPermissions
+	if len(opts.TargetNamespaces) == 1 && opts.TargetNamespaces[0] == "" {
+		for _, p := range rv1.CSV.Spec.InstallStrategy.StrategySpec.Permissions {
+			p.Rules = append(p.Rules, rbacv1.PolicyRule{
+				Verbs:     []string{"get", "list", "watch"},
+				APIGroups: []string{corev1.GroupName},
+				Resources: []string{"namespaces"},
+			})
+			clusterPermissions = append(clusterPermissions, p)
+		}
+	}
+
+	//nolint:prealloc
+	var objs []client.Object
+	for _, permission := range clusterPermissions {
+		saName := saNameOrDefault(permission.ServiceAccountName)
+		name, err := opts.UniqueNameGenerator(fmt.Sprintf("%s-%s", rv1.CSV.Name, saName), permission)
+		if err != nil {
+			return nil, err
+		}
+		objs = append(objs,
+			GenerateClusterRoleResource(name, WithRules(permission.Rules...)),
+			GenerateClusterRoleBindingResource(
+				name,
+				WithSubjects(rbacv1.Subject{Kind: "ServiceAccount", Namespace: opts.InstallNamespace, Name: saName}),
+				WithRoleRef(rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: name}),
+			),
+		)
+	}
+	return objs, nil
+}
+
+func BundleServiceAccountGenerator(rv1 *convert.RegistryV1, opts Options) ([]client.Object, error) {
+	allPermissions := append(
+		rv1.CSV.Spec.InstallStrategy.StrategySpec.Permissions,
+		rv1.CSV.Spec.InstallStrategy.StrategySpec.ClusterPermissions...,
+	)
+
+	var objs []client.Object
+	serviceAccountNames := sets.Set[string]{}
+	for _, permission := range allPermissions {
+		serviceAccountNames.Insert(saNameOrDefault(permission.ServiceAccountName))
+	}
+
+	for _, serviceAccountName := range serviceAccountNames.UnsortedList() {
+		// no need to generate the default service account
+		if serviceAccountName != "default" {
+			objs = append(objs, GenerateServiceAccountResource(serviceAccountName, opts.InstallNamespace))
+		}
+	}
+	return objs, nil
+}
+
+func BundleCRDGenerator(rv1 *convert.RegistryV1, _ Options) ([]client.Object, error) {
+	//nolint:prealloc
+	var objs []client.Object
+	for _, crd := range rv1.CRDs {
+		objs = append(objs, crd.DeepCopy())
+	}
+	return objs, nil
+}
+
+func BundleResourceGenerator(rv1 *convert.RegistryV1, _ Options) ([]client.Object, error) {
+	//nolint:prealloc
+	var objs []client.Object
+	for _, res := range rv1.Others {
+		supported, namespaced := registrybundle.IsSupported(res.GetKind())
+		if !supported {
+			return nil, fmt.Errorf("bundle contains unsupported resource: Name: %v, Kind: %v", res.GetName(), res.GetKind())
+		}
+
+		obj := res.DeepCopy()
+		if namespaced {
+			obj.SetNamespace(res.GetNamespace())
+		}
+
+		objs = append(objs, obj)
+	}
+	return objs, nil
+}
+
+func saNameOrDefault(saName string) string {
+	if saName == "" {
+		return "default"
+	}
+	return saName
```

Comment:
Nit: New-ish stdlib function that can simplify this.
```suggestion
	return cmp.Or(saName, "default")
```